src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java

```diff
@@ -329,13 +330,54 @@ private void addInvalidDisposalDateError(List<HearingTypeItem> hearingTypeItems,
             return;
         }
 
-        Optional<DateListedTypeItem> hearingTypeItem = findHearingTypeItem(hearingTypeItems, disposalDate);
-
-        if (hearingTypeItem.isEmpty() || !YES.equals(hearingTypeItem.get().getValue().getHearingCaseDisposed())) {
+        if (!disposedHearingWithSameDisposalDateOfJurisdictionExists(hearingTypeItems, disposalDate)) {
             errors.add(String.format(DISPOSAL_DATE_HEARING_DATE_MATCH, jurCode));
         }
     }
 
+    /**
+     * Checks if hearing type items has any hearing type which is disposed
+     * and has the same disposal date with the jurisdiction disposal date.
+     * @param hearingTypeItems All hearings
+     * @param disposalDate Disposal date of Jurisdiction
+     * @return boolean  if true then hearing list has disposed hearing with the
+     *                  same disposal date of jurisdiction.
+     */
+    private boolean disposedHearingWithSameDisposalDateOfJurisdictionExists(
+            List<HearingTypeItem> hearingTypeItems, String disposalDate) {
+        if (ObjectUtils.isNotEmpty(hearingTypeItems)) {
```

Comment:
is it worth inverting this condition to reduce complexity? not a big issue but just in terms of readability
if (CollectionUtils.isEmpty(hearingTypeItems) {
     return false;
 }